src/client/components/DicePanel.tsx

```diff
@@ -1,57 +1,382 @@
+    }
+  }, [myself.diceTemplateCategories, active]);
+
+  function addTemplateCategory() {
+    const freeIcon = userCategoryIcons.find((userIcon) => {
+      return !myself.diceTemplateCategories
+        .map(({ icon }: { icon: typeof categoryIcons[number] }) => icon)
+        .includes(userIcon);
+    });
+
+    if (!freeIcon) return;
+
+    dispatch({
+      actions: [
+        playerAddDiceTemplateCategory({
+          id: myself.id,
+          category: {
+            id: rrid<RRDiceTemplateCategory>(),
+            icon: freeIcon,
+            categoryName: "",
+            templates: [],
+          },
+        }),
+      ],
+      optimisticKey: "diceTemplateCategories",
+      syncToServerThrottle: DEFAULT_SYNC_TO_SERVER_DEBOUNCE_TIME,
+    });
+  }
+
+  const d20Part: RRDiceTemplatePartDice = {
+    id: rrid<RRDiceTemplatePart>(),
+    type: "dice",
+    faces: 20,
+    count: 1,
+    negated: false,
+    modified: "none",
+    damage: { type: null },
+  };
+  const characters = useServerState((state) => state.characters);
+  const character = characters.entities[myself.mainCharacterId!];
+
+  function getSavingThrowTemplates() {
+    const templates = [];
+
+    templates.push(
+      ...characterStatNames.map(
+        (statName: typeof characterStatNames[number]) => {
+          const proficiency =
+            character === undefined ? 0 : character.savingThrows[statName] ?? 0;
+
+          const parts: RRDiceTemplatePart[] = [
+            d20Part,
+            {
+              id: rrid<RRDiceTemplatePart>(),
+              type: "linkedStat",
+              name: statName,
+              damage: { type: null },
+            },
+          ];
+
+          if (proficiency !== 0)
+            parts.push({
+              id: rrid<RRDiceTemplatePart>(),
+              type: "linkedProficiency",
+              damage: { type: null },
+              proficiency,
+            });
+
+          return {
+            id: rrid<RRDiceTemplate>(),
+            name: `${statName} Saving Throw`,
+            notes: "",
+            parts,
+            rollType: null,
+          };
+        }
+      )
+    );
+    return templates;
+  }
+
+  function getSkillTemplates() {
+    const templates: RRDiceTemplate[] = [];
+
+    templates.push(
+      ...skillNames.map((skill) => {
+        const proficiency =
+          character === undefined ? 0 : character.skills[skill] ?? 0;
+        const parts: RRDiceTemplatePart[] = [
+          d20Part,
+          {
+            id: rrid<RRDiceTemplatePart>(),
+            type: "linkedStat",
+            name: skillMap[skill],
+            damage: { type: null },
+          },
+        ];
+        if (proficiency !== 0)
+          parts.push({
+            id: rrid<RRDiceTemplatePart>(),
+            type: "linkedProficiency",
+            damage: { type: null },
+            proficiency,
+          });
+
+        return {
+          id: rrid<RRDiceTemplate>(),
+          name: skill,
```

Comment:
Hmm, would `${skill} Check` be too verbose?

Reply:
I think at least in the Notification it has a nicer ring without it: "xx rolled for Athletics..."

Reply:
Fair point!